src/lib/homestore/shard_manager.cpp

```diff
@@ -20,4 +124,74 @@ ShardManager::Result< InfoList > HSHomeObject::_list_shards(pg_id pg) const {
     return folly::makeUnexpected(ShardError::UNKNOWN_PG);
 }
 
+void HSHomeObject::on_shard_message_commit(int64_t lsn, sisl::blob const& header, sisl::blob const& key,
+                                           void* user_ctx) {
```

Comment:
this will be called in `on_commit` of statemachine , right? what will we do in `on_precommit`

Reply:
on_precommit will call ReplDev to  alloc_blk() and it gives HeapChunkSelector a chance to bind a shard with the chunk.

Reply:
on_precommit and on_rollback is already added now. mostly their jobs is aligned with HomeObject design doc to write    ShardHeader to the HomeStore with BlkId returned, by that time the newly shard is binding with the specified chunk.